yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/main.nr

```diff
@@ -281,6 +285,126 @@ contract NonNativeToken {
         context.finish()
     }
 
+    // Shield creates a way for a user to move tokens from the public context into the private context.
+    open fn shield(
+        inputs: PublicContextInputs,
+        amount: Field,
+        secretHash: Field,
+    ) {
+        let storage = Storage::init();
+        let public_balances = storage.public_balances;
+
+        // Decrease user's balance.
+        let sender = inputs.call_context.msg_sender;
+        let sender_balance = public_balances.at(sender);
+        let current_sender_balance: Field = sender_balance.read();
+
+        if (current_sender_balance as u120) > (amount as u120) {
+            // User has sufficient balance so we decrement it by `amount`
+            let _void1 = sender_balance.write(current_sender_balance - amount);
+        }  
+        
+        // Create a commitment to the amount
+        let note = TransparentNote::new(amount, secretHash);
```

Comment:
if `current_sender_balance < amount`, we still seem to create a note?

Reply:
See #915